### PR TITLE
New color.map configurations for packages belonging to a user set in /etc/portage/sets/

### PR DESCRIPTION
**Pull request by armills**
https://bugs.gentoo.org/show_bug.cgi?id=577720

### CODE REVIEW

File: pym/_emerge/resolver/output.py [hunk 1]
```diff
@@ -699,22 +705,28 @@ def do_changelog(self, pkg, pkg_info):
...
 		@param pkg: _emerge.Package.Package instance
-		@rtype system and world booleans
+		@rtype set, system, and world booleans
 		"""
 		root_config = self.conf.roots[pkg.root]
-		system_set = root_config.sets["system"]
-		world_set  = root_config.sets["selected"]
-		system = False
-		world = False
+		set = None
```
Comment (Arfrever) [inline pym/_emerge/resolver/output.py]: `set` is a built-in class in Python. Do not name local variables in such way. I suggest `user_set`.

File: pym/_emerge/resolver/output_helpers.py [hunk 2]
```diff
@@ -612,12 +613,13 @@ def __init__(self):
 		self.previous_pkg = None
 		self.repo_path_real = ''
 		self.repo_name = ''
+		self.set = None
```
Comment (Arfrever) [inline pym/_emerge/resolver/output_helpers.py]: I suggest `self.user_set`.

File: pym/portage/output.py [hunk 3]
```diff
@@ -135,13 +135,16 @@ def color(fg, bg="default", attr=["normal"]):
 _styles["PKG_BLOCKER"]             = ( "red", )
 _styles["PKG_BLOCKER_SATISFIED"]   = ( "darkblue", )
 _styles["PKG_MERGE"]               = ( "darkgreen", )
+_styles["PKG_MERGE_SET"]           = ( "darkgreen", )
```
Comment (Arfrever) [inline pym/portage/output.py]: I suggest `PKG_MERGE_USER_SET` and similar changes below.

**Comment by armills** [comment]
Agreed, that's a better name.

**Comment by armills** [comment]
I pushed an update that gets rid of the hardcoded class name check. These commits can also be squashed once it's ready to merge.

**Comment by alexander-b** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Please send patches to the Portage mailing list[0] using git
send-email. This GitHub repository is simply a mirror.

[0]  <gentoo-portage-dev@lists.gentoo.org>
- -- 
Alexander
bernalex@gentoo.org
https://secure.plaimi.net/~alexander
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iQIcBAEBCgAGBQJW7WzmAAoJENQqWdRUGk8B6vIP/RYCYEf0Y0bHPT1w9jYM4/IW
uRKO9ucU4VBpxqvMpZ20gFK6gsil8G5SjrWztFqWH0iy4HXvYMFeQnihKbGK8XAr
A0f/X46NKdyWyhA9ti6i0BbwoXHwV+MNJVPCacdeCPtVH46O8LSnrwetjGvEMmCG
VKS4QZxfscTWqAVo3P9eR78qSd1CZCyUMmU6kUB4rAO7kPDDWE6gy6k0JhC4W6pM
Do9TZi/PAZnHNyxH//PXf/CxitaYAL4opz5Np2CFYNrE3jkAkMbr/KFrO7r+uaaJ
Qs+VZbH56fxEyfkg38Jn8CIl277p0MTrNLIp7CTYcx1fzSq1dMJYg5J7jou81Z4o
7w2UkOkJ5I+NQnn22NjG2TDRpY6MTMLSU1pdzLqbmEmU5PZghZ4X2xGfuNpose/+
GYsybo1we1KGRfnuNd4+TS2uS/dZBqjL0S3FYbAP4k7u2d1GxWQgoiCxJ2LyQDhV
o3eP+v4YpGMMhfCecJ1/zFgLSX/9xuKhPYVB0rPaj513p58cmOluhmPOJ8oioxP1
YtCkK8hCX6w7zv6fmDd6Uo/NjU0gAF+hLFaCviSDWY8uy6xfIDdl90+pmHBUY5N4
hqbk/Y/C5fMY+ise0sM7uEzFaX9T93sbCwJEqAaBXLHmeEsLjl8/BqitQc1PJOQZ
uYe3ApozsbZ6fpPUIEYN
=PERq
-----END PGP SIGNATURE-----

**Comment by armills** [comment]
https://archives.gentoo.org/gentoo-portage-dev/message/ba838c140b567bafaaf132bbabc39940

**Pull request closed without merging**